components/Wallet.tsx

```diff
@@ -0,0 +1,331 @@
+import { useCallback, FC, useRef, useEffect, useState } from "react";
+import { css } from "@emotion/react";
+import { Divider, Theme, LinearProgress } from "@mui/material";
+import { useCENNZWallet } from "@/providers/CENNZWalletProvider";
+import AccountIdenticon from "@/components/shared/AccountIdenticon";
+import { useCENNZExtension } from "@/providers/CENNZExtensionProvider";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import getTokenLogo from "@/utils/getTokenLogo";
+import { CENNZ_ASSET_ID, CPAY_ASSET_ID } from "@/constants";
+import { useWalletProvider } from "@/providers/WalletProvider";
+import { useSelectedAccount } from "@/hooks";
+
+const Wallet: FC = () => {
+	const { setWalletOpen, walletOpen, selectedWallet, setSelectedWallet } =
+		useWalletProvider();
+	const { balances, selectAccount, disconnectWallet } = useCENNZWallet();
+	const selectedAccount = useSelectedAccount();
+	const { accounts } = useCENNZExtension();
+
+	const onWalletDisconnect = useCallback(() => {
+		setWalletOpen(false);
+		setSelectedWallet(null);
+
+		if (selectedWallet === "CENNZnet") disconnectWallet();
+	}, [disconnectWallet, setWalletOpen, setSelectedWallet, selectedWallet]);
+
+	const ref = useRef<HTMLDivElement>();
+	const [balanceListHeight, setBalanceListHeight] = useState<number>(0);
+
+	useEffect(() => {
+		if (!walletOpen) return;
+		const setListHeight = () => {
+			const balanceList = ref.current;
+			if (!balanceList || !balances?.length) return setBalanceListHeight(0);
+			const rect = balanceList.getBoundingClientRect();
+			setBalanceListHeight(rect.height);
+		};
+
+		const id = setTimeout(setListHeight, 500);
+		setListHeight();
+
+		return () => clearTimeout(id);
+	}, [balances, walletOpen]);
+
+	const onAccountSelect = useCallback(
+		(event) => {
+			setBalanceListHeight(0);
+			selectAccount(
+				accounts.find(({ address }) => event.target.value === address)
+			);
+		},
+		[accounts, selectAccount]
+	);
+
+	if (!selectedAccount) return null;
+
+	return (
+		<div css={styles.modalContent}>
+			<div css={styles.accountHeader}>
+				<AccountIdenticon
+					value={selectedAccount.address}
+					theme="beachball"
+					size={50}
+				/>
+				<div css={styles.accountDetails}>
+					<div css={styles.accountName}>
+						{selectedAccount?.meta?.name ?? "METAMASK"}
+					</div>
+					<div
+						css={styles.accountAddress}
+						onClick={() =>
+							navigator.clipboard.writeText(selectedAccount.address)
+						}
+					>
+						{selectedAccount.address
```

Comment:
for `"MetaMask"` wallet, we want to show the Eth type address to avoid confusion